ably/realtime/connectionmanager.py

```diff
@@ -162,13 +175,26 @@ async def close_impl(self) -> None:
         self.cancel_suspend_timer()
         self.start_transition_timer(ConnectionState.CLOSING, fail_state=ConnectionState.CLOSED)
         if self.transport:
-            await self.transport.dispose()
+            # Try to send protocol CLOSE message to server for clean shutdown
+            # If connection is already broken, continue with dispose anyway
+            try:
+                await self.transport.close()
+            except Exception as e:
+                log.debug(f'ConnectionManager.close_impl(): failed to send CLOSE message: {e}')
+            await self.transport.dispose()  # Dispose transport resources
```

Comment:
It's not following the spec. Right now we send `CLOSE` protocol message and immediately after that we dispose connection, according to the spec we should wait for `CLOSED` protocol message.

Reply:
we still need to wait for the closed protocol message before actually disposing connection according to the spec:

>[(RTN12a)](https://sdk.ably.com/builds/ably/specification/main/features/#RTN12a) If the connection state is CONNECTED, sends a CLOSE ProtocolMessage to the server, transitions the state to CLOSING and waits for a CLOSED ProtocolMessage to be received

last part about "CLOSED ProtocolMessage to be received" requires websocket connection to live a bit longer 

but let's keep it out of the scope of this PR